OAK-Marketplace-Emotion-Recognition/emotion_analyzer.py

```diff
@@ -0,0 +1,284 @@
+
+    y1o, y2o = max(0, -y), min(img_overlay.shape[0], img.shape[0] - y)
+    x1o, x2o = max(0, -x), min(img_overlay.shape[1], img.shape[1] - x)
+
+    if y1 >= y2 or x1 >= x2 or y1o >= y2o or x1o >= x2o:
+        return
+
+    img_crop = img[y1:y2, x1:x2]
+    img_overlay_crop = img_overlay[y1o:y2o, x1o:x2o]
+    alpha = alpha_mask[y1o:y2o, x1o:x2o, np.newaxis]
+    alpha_inv = 1.0 - alpha
+    img_crop[:] = alpha * img_overlay_crop + alpha_inv * img_crop
+
+
+class EmotionAnalyzer:
+    def __init__(self, out_statistic_path: str = "result_statistic.json",
+                 visualization_shape: int = 300,
+                 report_statistics_shape: int = 720,
+                 emotion_bar_offset_percent: float = 0.1,
+                 emotion_images_path: str = "images/emoji"):
+        self.emotion_bar_offset_percent = emotion_bar_offset_percent
+        self.output_statistic_path = out_statistic_path
+        self.bar_size, self.padding, emotion_size = self.relative_size_estimation(visualization_shape)
+        self.vis_shape = visualization_shape
+        self.report_statistics_shape = report_statistics_shape
+        self.bars = {
+            "happy": EmotionBar(emotion='happy', size=self.bar_size,
+                                emotion_size=emotion_size,
+                                emotion_path= os.path.join(emotion_images_path, 'happy.png')),
+            "sad": EmotionBar(emotion='sad', size=self.bar_size, emotion_size=emotion_size,
+                              emotion_path= os.path.join(emotion_images_path, 'sad.png')),
+            "anger": EmotionBar(emotion='anger', size=self.bar_size, emotion_size=emotion_size,
+                                emotion_path= os.path.join(emotion_images_path, 'anger.png')),
+            "surprise": EmotionBar(emotion='surprise', size=self.bar_size, emotion_size=emotion_size,
+                                   emotion_path= os.path.join(emotion_images_path, 'surprise.png')),
+            "neutral": EmotionBar(emotion='neutral', size=self.bar_size, emotion_size=emotion_size,
+                                  emotion_path= os.path.join(emotion_images_path, 'neutral.png'))
+        }
+        self.fill_bars()
+
+    def relative_size_estimation(self, visualization_size, bar_amount: int = 5):
+        bar_section_size = visualization_size / bar_amount
+        bar_size = int(bar_section_size - 2 * bar_section_size * self.emotion_bar_offset_percent)
+        padding = int(bar_section_size - bar_size)
+        emotion_size = (int(bar_size / 3), int(bar_size / 3))
+        return bar_size, padding,  emotion_size
+
+    def fill_bars(self):
+        for x in self.bars.values():
+            x.update_bar()
+
+    def get_emotions_total_amount(self):
+        return sum([self.bars[x].update_count for x in self.bars.keys()])
+
+    def update_bars(self, emotion):
+        self.bars[emotion].update_count = self.bars[emotion].update_count + 1
```

Comment:
Don't need to it here. Read comments about EmotionBar. We can do it there in update_bar() function

Reply:
I get an idea here. We should probably keep counts inside EmotionAnalyzer for each Bar. What do you think?

Reply:
Yeah, probably it would be better